core/app/models/spree/shipment.rb

```diff
@@ -262,6 +262,10 @@ def shipping_method
       selected_shipping_rate.try(:shipping_method) || shipping_rates.first.try(:shipping_method)
     end
 
+    def shipment_address
+      self.try(:address) || order.ship_address
```

Comment:
@JDutil There is a BC issue in that when splitting to a new shipment it doesn't set a delivery address on the new shipment. @reinaris has fixed this but for BC reasons we should assume you can have a nil address and therefore must inherit the order's shipping address. An alternative (and maybe better approach) is to add a migration to set any missing addresses to the default shipping address.

Also, @reinaris will be submitting some changes to this PR soon that effectively make addresses immutable from the admin (by editing you create a new one). This will ensure data sanity when editing addresses that were originally shared with multiple shipments (without cloning them originally). We have also changed the edit address in the admin a little to make this easier to understand.